Sources/Filters/Sources/Arrow2DSource/index.js

```diff
@@ -0,0 +1,222 @@
+  const dataset = vtkPolyData.newInstance();
+
+  const points = new macro.TYPED_ARRAYS[model.pointType](4 * 3);
+
+  const thickOp = (model.height / 3) * model.thickness;
+
+  const offsetOp = model.height / 3 - thickOp;
+
+  const centerOffsetOp =
+    (model.height - offsetOp - (model.height / 3 - thickOp - offsetOp)) *
+    (1 - model.center);
+
+  points[0] = (model.width / 2) * -1;
+  points[1] = 0.0 - offsetOp - centerOffsetOp;
+  points[2] = 0.0;
+
+  points[3] = 0.0;
+  points[4] = model.height - offsetOp - centerOffsetOp;
+  points[5] = 0.0;
+
+  points[6] = model.width / 2;
+  points[7] = 0.0 - offsetOp - centerOffsetOp;
+  points[8] = 0.0;
+
+  points[9] = 0.0;
+  points[10] = model.height / 3 - thickOp - offsetOp - centerOffsetOp;
+  points[11] = 0.0;
+
+  const cells = Uint8Array.from([3, 0, 1, 3, 3, 1, 2, 3]);
+
+  vtkMatrixBuilder
+    .buildFromRadian()
+    .translate(...model.origin)
+    .rotateFromDirections([0, 1, 0], model.direction)
+    .apply(points);
+
+  dataset.getPoints().setData(points, 3);
+  dataset.getPolys().setData(cells, 1);
+
+  return dataset;
+}
+
+function vtkTriangleSource(publicAPI, model) {
+  const dataset = vtkPolyData.newInstance();
+
+  const points = new macro.TYPED_ARRAYS[model.pointType](3 * 3);
+
+  const centerOffsetOp = model.height * (1 - model.center);
+
+  points[0] = (model.width / 2) * -1;
+  points[1] = 0.0 - centerOffsetOp;
+  points[2] = 0.0;
+
+  points[3] = 0.0;
+  points[4] = model.height - centerOffsetOp;
+  points[5] = 0.0;
+
+  points[6] = model.width / 2;
+  points[7] = 0.0 - centerOffsetOp;
+  points[8] = 0.0;
+
+  const cells = Uint8Array.from([3, 0, 1, 2]);
+
+  vtkMatrixBuilder
+    .buildFromRadian()
+    .translate(...model.origin)
+    .rotateFromDirections([0, 1, 0], model.direction)
+    .apply(points);
+
+  dataset.getPoints().setData(points, 3);
+  dataset.getPolys().setData(cells, 1);
+
+  return dataset;
+}
+
+function vtkArrow2DSource(publicAPI, model) {
+  const shapeToSource = {
+    [ShapeType.TRIANGLE]: vtkTriangleSource,
+    [ShapeType.STAR]: vtkStarSource,
+    [ShapeType.ARROW_4]: vtk4PointsArrow,
+    [ShapeType.ARROW_6]: vtk6PointsArrow,
+  };
+
+  publicAPI.requestData = (inData, outData) => {
+    outData[0] = shapeToSource[model.shape](publicAPI, model);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+/* DEFAULT VALUES
+ * height modifies the height of the source , implemented in triangle 4PA 6PA
+ 
+ * width modifies the width of the source, implemented in triangle 4PA 6PA
+ 
+ * thickness modifies the shape of the source, 0 normal thickness 1 max
+ *		thickness, implemented in 4PA 6PA, scalar 
+ 
+ * center modifies the postion of the end of the Polyline in comparison to the
+ *		source, 1 the top of the arrow is on the limit of the line, 0 the bottom
+ *		of the arrow touches the line
+
+*/
```

Comment:
I think that documentation should be in api.md

Reply:
Doc in the code is fine too.